fuel-vm/src/storage.rs

```diff
@@ -105,12 +105,20 @@ macro_rules! double_key {
 
             /// Returns the reference to the first sub-key.
             pub fn $first_getter(&self) -> &$first {
-                unsafe { $first::as_ref_unchecked(&self.0[0..Self::first_end()]) }
+                $first::from_bytes_ref(
```

Comment:
Why, in some places, do we continue to use `unsafe` but in some, `.try_into().expect()`?

The `from_bytes_ref` uses `unsafe` under the hood, so why we can't use it here directly, if the macro guaranty that the layout of bytes is right for this type?

Reply:
The point is to contain the unsafe usage to as few places as possible. The `try_into()` adds no performance penalty here.